index.html

```diff
@@ -122,6 +177,196 @@ <h2>Methods</h2>
+    </p>
+    <p>
+      In the algorithm for generating subsequent offers in [[RTCWEB-JSEP]] section 5.2.2, replace "The
+      RTP header extensions MUST only include those that are present in the most recent answer"
+      with "For each RTP header extension listed in [[\HeaderExtensionsToOffer]] that is
+      also present in the most recent answer, and where direction is not "stopped", generate
+      an appropriate "a=extmap" line with "direction" set according to the rules of [[RFC5285]]
+      section 6, considering the direction in [[\HeaderExtensionsToOffer]] to indicate the
+      answerer's desired usage". This preserves the property that the set of extensions
+      may not grow with subsequent offers.
+    </p>
+    <p>
+      In the algorithm for generating initial answers in [[RTCWEB-JSEP]] section 5.3.1, replace "For
+      each supported RTP header extension that is present in the offer" with "For each
+      supported RTP header extension that is present in the offer and is also present in
+      [[\HeaderExtensionsToOffer]] with a direction different from "stopped"".
+    </p>
+    <p class="note">
+      Since JSEP does not know about WebRTC internal slots, merging this change requires
+      more work on a JSEP revision.
+    </p>
+    <p>
+      In the algorithm to <a>set an RTCSessionDescription</a>, add a bullet in the sections
+      for processing answer and pranswer (bullet 3.2.8.1.7 and 3.2.8.2.6):
+      <ul>
+        <li>Set the transceiver's [[\HeaderExtensionsNegotiated]] slot to the list of
+          RTP header extensions from the answer.
+        </li>
+      </ul>
+    </p>
+    <section>
+      <h2>Methods</h2>
+      <dl>
+        <dt>setOfferedRtpHeaderExtensions</dt>
+        <dd>
+          Execute the following steps:
+          <ol>
+            <li>Let <var>extlist</var> be the list of extensions in the argument.</li>
+            <li>
+              For each extension <var>ext</var> in <var>extlist</var>, do the following:
+              <ol>
+                <li>If <var>ext</var>'s "uri" value is not present in
+                  [[\HeaderExtensionsToOffer]], throw a NotSupported error.
+                </li>
+                <li>
+                  Let <var>entry</var> be the entry with the same "uri" value in
+                  [[\HeaderExtensionsToOffer]].
+                </li>
+                <li>If <var>ext</var>'s "direction" field is not "sendrecv", and "uri"
+                  indicates a mandatory-to-use attribute that is required to be both
+                  sent and received, throw an IllegalArgument error.
+                </li>
+                <li>if <var>ext</var>'s "direction" field is "stopped", and "uri" indicates
+                  a mandatory-to-implement extension, throw an IllegalArgument error.
+                </li>
+                <li>
+                  Set the "direction" field of <var>entry</var> to <var>ext</var>'s
+                  "direction" field.
```

Comment:
This algorithm could modify one entry and throw an exception at the next entry. That would be a bug.

Make a copy of all the entries, and only if we manage to iterate all ext:s without throwing to we set [[\HeaderExtensionsToOffer]] as the last step before returning.